src/commissaire_service/transport/ansibleapi.py

```diff
@@ -0,0 +1,333 @@
+            'commissaire', 'data/ansible/playbooks/restart.yaml')
+        restart_command = " ".join(oscmd.restart())
+        return self._run(
+            ips, key_file, play_file, [0, 2],
+            {'commissaire_restart_command': restart_command},
+            disable_reconnect=True)
+
+    def get_info(self, ip, key_file):
+        """
+        Get's information from the host via ansible.
+
+        :param ip: IP address to check.
+        :type ip: str
+        :param key_file: Full path to the file holding the private SSH key.
+        :type key_file: str
+        :returns: A dictionary of facts for a Host model
+        :rtype: dict
+        :raises subprocess.CalledProcessError: if Ansible returns a non-zero
+                                               exit status
+        """
+        ansible_facts = gather_facts(ip, self._get_ansible_args(key_file))
+        self.logger.debug('Ansible facts: {0}'.format(ansible_facts))
+        facts = {}
+        facts['os'] = ansible_facts['ansible_distribution'].lower()
+        facts['cpus'] = ansible_facts['ansible_processor_cores']
+        facts['memory'] = ansible_facts['ansible_memory_mb']['real']['total']
+        space = 0
+        for x in ansible_facts['ansible_mounts']:
+            space += x['size_total']
+        facts['space'] = space
+
+        self.logger.debug('Grabbed Facts: {0}'.format(facts))
+        return facts
+
+    def check_host_availability(self, host, key_file):
+        """
+        Checks if a host node is available.
+
+        :param host: The host model to check.
+        :type host: commissaire.models.Host
+        :param key_file: The path to the ssh_priv_key.
+        :type key_file: str
+        :returns: An Ansible status code (0=Success)
+        :rtype: int
+        """
+        play_file = resource_filename(
+            'commissaire',
+            'data/ansible/playbooks/check_host_availability.yaml')
+        return self._run(
+            host.address, key_file, play_file, [0, 3], disable_reconnect=True)
+
+    def bootstrap(self, ip, key_file, oscmd, etcd_config, cluster, network):
+        """
+        Bootstraps a host via ansible.
+
+        :param ip: IP address to bootstrap.
+        :type ip: str
+        :param key_file: Full path to the file holding the private SSH key.
+        :type key_file: str
+        :param oscmd: OSCmd class to use
+        :type oscmd: commissaire_service.oscmd.OSCmdBase
+        :param etcd_config: An EtcdStoreHandler configuration
+        :type etcd_config: dict
+        :param cluster: A cluster model for the host
+        :type cluster: commissaire.models.Cluster
+        :param network: A network model for the host's cluster
+        :type network: commissaire.models.Network
+        :returns: An Ansible status code (0=Success)
+        :rtype: int
+        """
+        self.logger.debug('Using {0} as the oscmd class for {1}'.format(
+            oscmd.os_type, ip))
+
+        play_vars = {
```

Comment:
I can't wait till we port fully over so we no longer have to do this :moneybag: :moneybag: :moneybag: